lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/Transactions.hs

```diff
@@ -244,101 +238,152 @@ spec = do
-        let bs = block0H <> getHash (signData inps txOuts)
-        let (Just result) = isOwned st' (rootXPrv, pwd) addrD
-        let wits = [sign bs result]
-        return (Tx (fragmentId inps txOuts wits) inps txOuts, wits)
+    toRawBytes base bs = case convertFromBase base (T.encodeUtf8 bs) of
+        Left err -> error err
+        Right res -> res
 
-    encodeTx (tx, wits) msgType =
-        let encode ((Tx _ inps' outs'), wits') msgType' = runPut
-                $ withHeader msgType'
-                $ putSignedTx inps' outs' wits'
-        in T.decodeUtf8 $ convertToBase Base64 $ BL.toStrict $
-                encode (tx, wits) msgType
+
+encodeTx :: (Tx, [TxWitness]) -> MessageType -> Base -> Text
+encodeTx (tx, wits) msgType base =
+    T.decodeUtf8 $ convertToBase base $ BL.toStrict $ encode (tx, wits) msgType
+  where
+    encode ((Tx _ inps' outs'), wits') msgType' = runPut
+        $ withHeader msgType'
+        $ putSignedTx inps' outs' wits'
+
+data ExternalTxFixture = ExternalTxFixture
+    { srcWallet :: ApiWallet
+    , dstWallet :: ApiWallet
+    , feeMin :: Natural
+    , txWithWits :: (Tx, [TxWitness])
+    }
+
+fixtureExternalTx
+    :: forall t. (EncodeAddress t, DecodeAddress t)
+    => (Context t) -> Natural -> IO ExternalTxFixture
+fixtureExternalTx ctx toSend = do
+    -- we use faucet wallet as wSrc
+    let password = "cardano-wallet" :: Text
+    let mnemonicFaucet =
+            [ "vibrant", "orphan", "put", "metal", "wreck"
+            , "yellow", "final", "bacon", "matter", "spring"
+            , "stage", "enhance", "unaware", "skill", "fiber"
+            ] :: [Text]
+    let restoreFaucetWallet = Json [json| {
+            "name": "Faucet Wallet",
+            "mnemonic_sentence": #{mnemonicFaucet},
+            "passphrase": #{password}
+            } |]
+    r0 <- request
+        @ApiWallet ctx ("POST", "v2/wallets") Default restoreFaucetWallet
+    verify r0
+        [ expectResponseCode @IO HTTP.status202
+        , expectFieldEqual walletName "Faucet Wallet"
+        ]
+    let wSrc = getFromResponse Prelude.id r0
+    -- we take input by lookking at transactions of the faucet wallet
+    txsSrc <- listAllTransactions ctx wSrc
+    let (ApiTransaction (ApiT theTxId) _ _ _ _ _ outs _):_ = reverse txsSrc
+    let (AddressAmount ((ApiT addrSrc),_) (Quantity amt)):_ = NE.toList outs
+    let (rootXPrv, pwd, st) = getSeqState mnemonicFaucet password
+    -- we create change address
+    let (addrChng, st') = genChange pwd st
+    -- we generate address private keys for all source wallet addresses
+    let (Just keysAddrSrc) = isOwned st' (rootXPrv, pwd) addrSrc
+    let (Just keysAddrChng) = isOwned st' (rootXPrv, pwd) addrChng
+
+    -- we create destination empty wallet
+    let password1 = "Secure Passphrase" :: Text
+    let createWallet = Json [json| {
+            "name": "Destination Wallet",
+            "mnemonic_sentence": #{mnemonics15},
```

Comment:
Is it mandatory to use explicit `mnemonic15` as a destination wallet and then get the destination address from it? I'm wondering if we cannot just use `emptyWallet` method? 

The reason I'm asking is because when constructing tx using `jcli` you only need to provide destination address to the transaction being constructed (no key related to destination address is needed), so it seems the mnemonics don't have to be explicit (?)

Reply:
cannot do it. I use something like that 
```
let (rootXPrv1, pwd1, st1) = getSeqState mnemonics15 password1
let (Just keysAddrDest) = isOwned st1 (rootXPrv1, pwd1) addrDest'
```
in order to populate keystore. I need well defined (ie. have password and mnemonic) wallet in order to sign it . When playing manually you can not be so restrictive in your test